.github/workflows/rust.yml

```diff
@@ -30,6 +30,28 @@ jobs:
       - name: Run tests
         run: cargo test
 
+  push_package_demo:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3
+      - name: Login to GHCR
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+        # if: github.event_name != 'pull_request'
```

Comment:
I'm a little undecided on when to run this, I think every commit is _okay_, but then that means if we, for whatever reason, break the `shell.jsonnet` it'll break it for anyone else trying to use it.

Maybe that's not really a huge deal for now and is something we deal with when it comes to it.

I've commented this to raise this question (and to verify it can push into the right place from GitHub Actions), but before this merges/we decide what to do I'll rejig this first.